### PR TITLE
Feature/CV2-5319-trash-relay-article-refresh

### PR DESCRIPTION
**Pull request by sarahkeh**
## Description

Page no longer needs a full reload after an item is trashed.  After trashing an item, a refresh=true string is pushed to the url, triggering a refresh of the current viewed list.  In the Update function, counters used for the sidebar are returned which update the values to reflect the moved article.  This was done by making every counter explicitly named, removing the middle ArticleCoreListCounter component, and moving the query for those values to the parent ArticleDrawer queryRenderer.  Also renamed ProjectsListCounter to DrawerListCounter to indicate a more generic use.  

References: CV2-5319

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Trashed and restored Fact Checks, Explainers, and Published fact-checks.  

## Things to pay attention to during code review

If counters and lists aren't updating, check that you're in Production mode.   

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)
- [x] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed

### CODE REVIEW

File: src/app/components/drawer/DrawerArticles.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ const DrawerArticlesComponent = ({ team }) => {
               <div className={styles.listLabel}>
                 <FormattedMessage defaultMessage="Claim & Fact-Checks" description="Label for a list displayed on the left sidebar that includes items that have claim & fact-checks" id="articlesComponent.claimAndFactChecks" tagName="span" />
               </div>
-              <ArticleCoreListCounter teamSlug={team.slug} type="fact-check" />
+              <DrawerListCounter key={team.factChecksCount} numberOfItems={team.factChecksCount} />
```
Comment (amoedoamorim) [inline src/app/components/drawer/DrawerArticles.js:62]: These `key` props passed to the `DrawerListCounter` seem unnecessary as this isn't the output of a list.map and `DrawerListCounter` has no internal state that would need being reset upon key updating. Can you plz remove them?

File: src/app/components/article/ArticleTrash.js [hunk 2]
```diff
@@ -20,6 +21,11 @@ const updateExplainer = graphql`
         id
         trashed
       }
+      team {
+        articles_count
```
Comment (amoedoamorim) [inline src/app/components/article/ArticleTrash.js:25]: Is this generic `articles_count` still needed now that we have aliases for each list?

**Review by amoedoamorim — APPROVED** [review]
Please review my comments and apply them if they're pertinent, other than that. Looks pretty good, approving.

File: src/app/components/article/ArticleTrash.js (hunk 2 above)
Comment (sarahkeh) [reply src/app/components/article/ArticleTrash.js]: No, you're right, removed.

File: src/app/components/drawer/DrawerArticles.js (hunk 1 above)
Comment (sarahkeh) [reply src/app/components/drawer/DrawerArticles.js]: Removed

**Pull request merged**